app/lib/service/youtube/backend.dart

```diff
@@ -110,24 +113,33 @@ class _PkgOfWeekVideoFetcher {
 
     try {
       final videos = <PkgOfWeekVideo>[];
+      final videoIds = <String>{};
+      int pageIndex = 0;
       String? nextPageToken;
       for (var check = true; check && videos.length < 50;) {
         final rs = await youtube.playlistItems.list(
           ['snippet', 'contentDetails'],
           playlistId: powPlaylistId,
         );
-        videos.addAll(rs.items!.map(
-          (i) => PkgOfWeekVideo(
+        for (final i in rs.items!) {
+          final video = PkgOfWeekVideo(
             videoId: i.contentDetails!.videoId!,
             title: i.snippet!.title!,
             description:
                 (i.snippet?.description ?? '').trim().split('\n').first,
             thumbnailUrl: i.snippet!.thumbnails!.high!.url!,
-          ),
-        ));
+          );
+          if (videoIds.add(video.videoId)) {
+            videos.add(video);
+          } else {
+            _logger.warning(
+                'Duplicate videoId "${video.videoId}" on page $pageIndex.');
```

Comment:
Should we also log the previous page this occurred on?

Reply:
Do you think it is useful? We can't really use the information, not even the `pageIndex` much:
- the sort order is time descending, a new item will push the page indexes
- the pagination is based on continuation token.

Reply:
Hmm - do we then want to log pageIndex at all?

Only thing I was curious about was if the duplicates could happen on the same page or not....

Reply:
Ah, that's a good point!